autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,242 @@
+        """ Returns crude categorization of feature types """
+        return self.feature_types_metadata[feature_type] if feature_type in self.feature_types_metadata else []
+
+    def _get_solver(self):
+        if self.problem_type == BINARY:
+            solver = 'lbfgs'  # TODO use liblinear for smaller datasets
+        elif self.problem_type == MULTICLASS:
+            solver = 'saga'  # another option is lbfgs
+        else:
+            solver = 'lbfgs'
+        return solver
+
+    # TODO: handle collinear features - they will impact results quality
+    def preprocess(self, X: DataFrame, is_train=False, vect_max_features=1000):
+        X = X.copy()
+        feature_types = self._get_types_of_features(X)
+        if is_train:
+            self.preprocess_train(X, feature_types, vect_max_features)
+        X = self.pipeline.transform(X)
+
+        return X
+
+    def preprocess_train(self, X, feature_types, vect_max_features):
+        transformer_list = []
+        if len(feature_types['language']) > 0:
+            pipeline = Pipeline(steps=[
+                ("preparator", NlpDataPreprocessor(nlp_cols=feature_types['language'])),
+                ("vectorizer",
+                 TfidfVectorizer(ngram_range=self.params['proc.ngram_range'], sublinear_tf=True, max_features=vect_max_features, tokenizer=self.tokenize))
+            ])
+            transformer_list.append(('vect', pipeline))
+        if len(feature_types['onehot']) > 0:
+            pipeline = Pipeline(steps=[
+                ('generator', OheFeaturesGenerator(cats_cols=feature_types['onehot'])),
+            ])
+            transformer_list.append(('cats', pipeline))
+        if len(feature_types['continuous']) > 0:
+            pipeline = Pipeline(steps=[
+                ('generator', NumericDataPreprocessor(cont_cols=feature_types['continuous'])),
+                ('imputer', SimpleImputer(strategy=self.params['proc.impute_strategy'])),
+                ('scaler', StandardScaler())
+            ])
+            transformer_list.append(('cont', pipeline))
+        if len(feature_types['skewed']) > 0:
+            pipeline = Pipeline(steps=[
+                ('generator', NumericDataPreprocessor(cont_cols=feature_types['skewed'])),
+                ('imputer', SimpleImputer(strategy=self.params['proc.impute_strategy'])),
+                ('quantile', QuantileTransformer(output_distribution='normal')),  # Or output_distribution = 'uniform'
+            ])
+            transformer_list.append(('skew', pipeline))
+        self.pipeline = FeatureUnion(transformer_list=transformer_list)
+        self.pipeline.fit(X)
+
+    def _set_default_params(self):
+        default_params = {
+            'C': 1,
+            'vectorizer_dict_size': 75000,
+            'proc.ngram_range': (1, 5),
+            'proc.skew_threshold': 0.99,
+            'proc.impute_strategy': 'median',  # strategy argument of sklearn.SimpleImputer() used to impute missing numeric values
```

Comment:
Each of these hyperparameters should be explained with comments.

Also could you put these in a separate file located in a hyperparameters/ subdirectory like we have for the other (non-sklearn) models?  This would make it more standardized, and a lot easier for users to quickly view the  hyperparameters they can specify.

Should 'regression_option' be another hyperparameter in the 'LR' dict? Otherwise how can a user choose between 'ridge', 'lasso' without delving into undocumented Trainer code?  Also consider standardizing this hyperparameter with the 'penalty' option in LogisticRegression which allows you to choose between 'l1', 'l2'. I would call it something more informative like 'penalty' instead of 'regression_option' (and you can just appropriately translate the 'l1' & 'l2' options into Lasso & Ridge internally).

Finally, what are your thoughts of exposing additional hyperparameters that can be shared

Reply:
* Added comments
* Moved-out hyperparameters and searchspaces
* `regression_option` renamed into `penalty` passed as a hyper-parameter with options: `L1`/`L2`
* fit_intercept/solver/random_state can be passed through hyper-parameters already.